lib/registry.ex

```diff
@@ -72,8 +72,10 @@ defmodule Registry do
       # Prints #PID<...> where the pid is for the process that called register/3 above
       #=> :ok
 
-  Keep in mind dispatching happens in the process that calls `dispatch/3`, so
-  the callback is executed there.  The registered processes are not involved in
+  Keep in mind that if the registry is not partitioned then dispatching happens
```

Comment:
I think the point we need to make clear here is that the callback is *not* invoked in the registered process but rather in the caller (or linked to the caller). The previous section was confusing but I think we can trim this down.